File: app/src/test/java/ca/ualberta/taskr/TaskUnitTest.java
package ca.ualberta.taskr;

/**
 * Created by Jacob Bakker on 2/22/2018.
 */

import android.media.Image;

import com.mapbox.mapboxsdk.geometry.LatLng;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import java.io.File;
import java.util.ArrayList;

import ca.ualberta.taskr.models.Bid;
import ca.ualberta.taskr.models.Task;
import ca.ualberta.taskr.models.TaskStatus;

import static junit.framework.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class TaskUnitTest {
    public String owner = "Corkus";
    public String title = "I'm a wot";
    public TaskStatus status;
    public ArrayList<Bid> bids = new ArrayList<Bid>();
    public String description = "4 mana 7/7";
    public ArrayList<Image> photos = new ArrayList<Image>();
    public LatLng location;
    public String chosenBidder = "The Mask";

    public String newBidName1 = "Mr. MoneyBags McGee's Monetary Mmmm";
    public String newBidName2 = "A Bribe";
    public float newBidAmount1 = 0.01f;
    public float newBidAmount2 = 6.66f;

    @Test
    public void testGetOwner() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        String testOwner = task.getOwner();
        assertEquals(testOwner, owner);
    }

    @Test
    public void testGetTitle() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        String testTitle = task.getTitle();
        assertEquals(testTitle, title);
    }

    @Test
    public void testGetStatus() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        TaskStatus testStatus = task.getStatus();
        assertEquals(testStatus, status);
    }

    @Test
    public void testGetBids() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        ArrayList<Bid> testBids = task.getBids();
        assertEquals(testBids, bids);
    }

    @Test
    public void testAddAndGetBid() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        Bid newBid1 = new Bid(newBidName1, newBidAmount1);
        Bid newBid2 = new Bid(newBidName2, newBidAmount2);
        task.addBid(newBid1);
        task.addBid(newBid2);
        Bid testBid = task.getBidAtIndex(0);
        assertEquals(testBid, newBid1);
        testBid = task.getBidAtIndex(1);
        assertEquals(testBid, newBid2);
    }

    @Test
    public void testSetBidAtIndex() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        Bid newBid1 = new Bid(newBidName1, newBidAmount1);
        Bid newBid2 = new Bid(newBidName2, newBidAmount2);
        task.addBid(newBid1);
        task.setBidAtIndex(newBid2, 0);
        Bid testBid = task.getBidAtIndex(0);
        assertEquals(testBid, newBid2);
    }
    @Test
    public void testGetDescription() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        String testDescription = task.getDescription();
        assertEquals(testDescription, description);
    }

    @Test
    public void testGetPhotos() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        ArrayList<Image> testPhotos = task.getPhotos();
        assertEquals(testPhotos, photos);
    }

    @Test
    public void testAddAndGetPhoto() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        task.addPhoto(newPhoto1);
        Image testPhoto = task.getPhotoAtIndex(0);
        assertEquals(testPhoto, newPhoto1);
    }

    @Test
    public void testGetLocation() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        LatLng testLocation = task.getLocation();
        assertEquals(testLocation, location);
    }
    @Test
    public void testGetChosenBidder() {
        Task task = new Task(owner, title, status, bids, description, photos, location, chosenBidder);
        String testChosenBidder = task.getChosenBidder();
        assertEquals(testChosenBidder, chosenBidder);
    }
}

File: app/src/test/java/ca/ualberta/taskr/models/TaskUnitTest.kt
package ca.ualberta.taskr.models

/**
 * Created by Jacob Bakker on 2/22/2018.
 */

import android.media.Image

import com.mapbox.mapboxsdk.geometry.LatLng

import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

import java.util.ArrayList

import ca.ualberta.taskr.models.Bid
import ca.ualberta.taskr.models.Task
import ca.ualberta.taskr.models.TaskStatus

import org.junit.Assert.assertEquals

@RunWith(RobolectricTestRunner::class)
class TaskUnitTest {
    private var owner = "Corkus"
    private var title = "I'm a wot"
    private var status: TaskStatus? = null
    private var bids = ArrayList<Bid>()
    private var description = "4 mana 7/7"
    private var photos = ArrayList<String>()
    private var location: LatLng? = null
    private var chosenBidder = "The Mask"

    private var newBidName1 = "Mr. MoneyBags McGee's Monetary Mmmm"
    private var newBidName2 = "A Bribe"
    private var newBidAmount1 = 0.01f
    private var newBidAmount2 = 6.66f


    /**
     * Test getting the owner from a task
     */
    @Test
    fun testGetOwner() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(owner, task.owner)
    }

    /**
     * Test getting the title from a task
     */
    @Test
    fun testGetTitle() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(title, task.title)
    }

    /**
     * Test getting the status of a task
     */
    @Test
    fun testGetStatus() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(status, task.status)
    }

    /**
     * test getting the bids from a task
     */
    @Test
    fun testGetBids() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(bids, task.bids)
    }

    /**
     * Test adding a bid to a task
     */
    @Test
    fun testAddAndGetBid() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        val newBid1 = Bid(newBidName1, newBidAmount1.toDouble(), false)
        val newBid2 = Bid(newBidName2, newBidAmount2.toDouble(), false)
        task.addBid(newBid1)
        task.addBid(newBid2)
        var testBid = task.getBidAtIndex(0)
        assertEquals(newBid1, testBid)
        testBid = task.getBidAtIndex(1)
        assertEquals(newBid2, testBid)
    }

    /**
     * test updating a bid at an index
     */
    @Test
    fun testSetBidAtIndex() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        val newBid1 = Bid(newBidName1, newBidAmount1.toDouble(), false)
        val newBid2 = Bid(newBidName2, newBidAmount2.toDouble(), false)
        task.addBid(newBid1)
        task.setBidAtIndex(newBid2, 0)
        val testBid = task.getBidAtIndex(0)
        assertEquals(newBid2, testBid)
    }

    /**
     * Test getting the task description
     */
    @Test
    fun testGetDescription() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(description, task.description)
    }

    /**
     * test getting the tasks photos
     */
    @Test
    fun testGetPhotos() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(photos, task.photos)
    }

    /**
     * test getting a tasks location
     */
    @Test
    fun testGetLocation() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(location, task.location)
    }

    /**
     * test getting a tasks chosen bidder
     */
    @Test
    fun testGetChosenBidder() {
        val task = Task(owner, title, status, bids, description, photos, location, chosenBidder)
        assertEquals(chosenBidder, task.chosenBidder)
    }

    /**
     * test completing and closing a task
     */
    @Test
    fun testCloseTask(){}

    /**
     * test reopening the task
     */
    @Test
    fun testReopenTask(){}

    /**
     * test seeing if a task if open for bidding
     */
    @Test
    fun testIsTaskOpenForBidding(){}

}
